### ISSUE TITLE
基础API和高层API问题

### ISSUE THREAD

**Issue by yinpu** [PaddlePaddle/Paddle#36650]
1.高层API中，fit中使用EarlyStopping的callback，如果只想保存最好的模型，不保存其他模型，该怎么做？
2.基础API训练中（不使用fit，采用loss.backward()+optimizer.step()方式训练），如果在某些layer的ParamAttr设置了regularizer，怎么在loss中添加对应的regularizer？如果不能，ParamAttr只能在fit方式有用吗？

**Comment by LiuChiachi** [comment]
您好！
1.fit的参数`save_freq`设大一点，超过原本的step数即可，EarlyStopping里`save_best_model`默认选True就可以了；
2. @chenwhql 可以帮忙看一下基础API下ParamAttr的使用吗～感谢

**Comment by LiuChiachi** [comment]
2.可以参考`ParamAttr` 文档中`regularizer`的释义：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/ParamAttr_cn.html#paramattr

**Comment by yinpu** [comment]
> 2.可以参考`ParamAttr` 文档中`regularizer`的释义：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/ParamAttr_cn.html#paramattr

比如model中某些层已经设置好了ParamAttr，训练的代码如下：

pred = model(x)
loss = loss_fn(pred, y)   #此处loss_fn为mse_loss或者ce_loss损失函数
loss.backward()
optimizer.step()

这样optimizer优化的是loss，还是loss+regularizer loss(L1 loss或者L2 loss)

**Comment by LiuChiachi** [comment]
会在参数上加正则项，和loss没有直接关系。
比如之前优化的是参数w，加上正则之后会优化w + regularizer(w)
